pymess/backend/dialer/__init__.py

```diff
@@ -0,0 +1,186 @@
+        """
+        raise NotImplementedError
+
+    def publish_messages(self, messages):
+        """
+        Sends more dialer messages together. If concrete dialer backend provides send more messages at once the method
+        can be overridden
+        :param messages: list of dialer messages
+        """
+        [self.publish_message(message) for message in messages]
+
+    def send(self, recipient, content, related_objects=None, tag=None, template=None, **dialer_kwargs):
+        """
+        Send dialer message with the text content to the phone number (recipient)
+        :param recipient: phone number of the recipient
+        :param content: text content of the message
+        :param related_objects: list of related objects that will be linked with the SMS message using generic
+        relation
+        :param tag: string mark that will be saved with the message
+        :param template: template object from which content of the message was create
+        :param dialer_kwargs: extra attributes that will be stored to the message
+        """
+        message = self.create_message(recipient, content, related_objects, tag, template, **dialer_kwargs)
+        self.publish_message(message)
+        return message
+
+    def bulk_send(self, recipients, content, related_objects=None, tag=None, template=None, **dialer_kwargs):
+        """
+        Send more dialer messages in one bulk
+        :param recipients: list of phone numbers of recipients
+        :param content: text content of the messages
+        :param related_objects: list of related objects that will be linked with the SMS message using generic
+        relation
+        :param tag: string mark that will be saved with the message
+        :param template: template object from which content of the message was create
+        :param dialer_kwargs: extra attributes that will be stored with messages
+        """
+        messages = [
+            self.create_message(recipient, content, related_objects, tag, template, **dialer_kwargs)
+            for recipient in recipients
+        ]
+        self.publish_messages(messages)
+        return messages
+
+    def get_initial_dialer_state(self, recipient):
+        """
+        returns initial state for logged dialer instance.
+        :param recipient: phone number of the recipient
+        """
+        return DialerMessage.STATE.UNKNOWN
+
+    def _update_dialer_states(self, messages):
+        """
+        If dialer sender provides check dialer delivery this method can be overridden.
+        :param messages: messages which state will be updated
+        """
+        raise NotImplementedError('Check dialer state is not supported with the backend')
+
+    def bulk_check_dialer_states(self):
+        """
+        Method that find messages that is not in the final state and updates its states.
+        """
+        messages_to_check = DialerMessage.objects.exclude(state=DialerMessage.STATE.CLOSED)
```

Comment:
a DONE nejsou ty ukončené?

Reply:
`DONE` zatial zamyslam ako 'hovor vykonany'. `CLOSED` ako vymahanie skoncene. alebo to neni takto potrebne?

asi hodim do diskusie pod taskom tie stavy...

Reply:
@rubickcz nerozumiem tvojim otazkam